synapse/handlers/presence.py

```diff
@@ -408,6 +433,20 @@ async def process_replication_rows(self, token, rows):
         stream_id = token
         await self.notify_from_replication(states, stream_id)
 
+        # Handle poking the local federation sender, if there is one.
+        if not self._federation:
+            return
+
+        hosts_and_states = await get_interested_remotes(
+            self.store,
+            self.presence_router,
+            states,
+            self.state,
+        )
+
+        for destinations, states in hosts_and_states:
+            self._federation.send_presence_to_destinations(states, destinations)
```

Comment:
I can't help but feel that having four separate instances of the `get_interested_remotes; send_presence_to_destinations` pattern suggests that we should do something slightly different.

is there any milage in replacing `get_interested_remotes` with a `send_presence_to_interested_destinations`?